File: src/main/java/gargoyle/netsquares/util/Rnd.java
package gargoyle.netsquares.util;

import java.util.Random;


public final class Rnd {
    private static final Random r = new Random();

    private Rnd() {
    }

    public static int rnd(int a, int b) {
        return r.nextInt(b - a + 1) + a;
    }

    private static <T> T rnd(T[] array) {
        return array[r.nextInt(array.length)];
    }

    public static <T extends Enum<T>> T rnd(Class<T> enumClass) {
        return rnd(enumClass.getEnumConstants());
    }
}

File: src/main/kotlin/gargoyle/netsquares/util/Rnd.kt
package gargoyle.netsquares.util

import java.util.Random
import kotlin.reflect.KClass

object Rnd {
    private val r = Random()
    fun rnd(a: Int, b: Int): Int {
        return r.nextInt(b - a + 1) + a
    }

    fun <T> rnd(enumClass: KClass<T>): T
            where T : Enum<T> {
        return rnd(enumClass.java.enumConstants)
    }

    private fun <T : Any> rnd(array: Array<T>): T {
        return array[r.nextInt(array.size)]
    }
}
